datastore/providers/mongodb_atlas_datastore.py

```diff
@@ -0,0 +1,265 @@
+MONGODB_INDEX = os.environ.get("MONGODB_INDEX", "default")
+OVERSAMPLING_FACTOR = 10
+MAX_CANDIDATES = 10_000
+UPSERT_BATCH_SIZE = 100
+
+
+class MongoDBAtlasDataStore(DataStore):
+
+    def __init__(
+        self,
+        atlas_connection_uri: str = MONGODB_CONNECTION_URI,
+        index_name: str = MONGODB_INDEX,
+        database_name: str = MONGODB_DATABASE,
+        collection_name: str = MONGODB_COLLECTION,
+        oversampling_factor: float = OVERSAMPLING_FACTOR,
+    ):
+        """
+        Initialize a MongoDBAtlasDataStore instance.
+
+        Parameters:
+        - index_name (str, optional): Vector search index. If not provided, default index name is used.
+        - database_name (str, optional): Database. If not provided, default database name is used.
+        - collection_name (str, optional): Collection. If not provided, default collection name is used.
+        - oversampling_factor (float, optional): Oversampling factor for data augmentation.
+                                                 Default is OVERSAMPLING_FACTOR.
+
+        Raises:
+        - ValueError: If index_name is not a valid string.
+
+        Attributes:
+        - index_name (str): Name of the index.
+        - database_name (str): Name of the database.
+        - collection_name (str): Name of the collection.
+        - oversampling_factor (float): Oversampling factor for data augmentation.
+        """
+
+        self.atlas_connection_uri = atlas_connection_uri
+        self.oversampling_factor = oversampling_factor
+        self.database_name = database_name
+        self.collection_name = collection_name
+
+        if not (index_name and isinstance(index_name, str)):
+            raise ValueError("Provide a valid index name")
+        self.index_name = index_name
+
+        # TODO: Create index via driver https://jira.mongodb.org/browse/PYTHON-4175
+        # self._create_search_index(num_dimensions=1536, path="embedding", similarity="dotProduct", type="vector")
+
+    @cached_property
+    def client(self):
+        return self._connect_to_mongodb_atlas(
+            atlas_connection_uri=MONGODB_CONNECTION_URI
+        )
+
+    async def _upsert(self, chunks: Dict[str, List[DocumentChunk]]) -> List[str]:
+        """
+        Takes in a list of document chunks and inserts them into the database.
+        Return a list of document ids.
+        """
+        documents_to_upsert = []
+        for chunk_list in chunks.values():
+            for chunk in chunk_list:
+                documents_to_upsert.append(
+                    self._convert_document_chunk_to_mongodb_document(chunk)
+                )
+        # Upsert documents into the MongoDB collection
+        logger.info(f"{self.database_name}: {self.collection_name}")
+        await self.client[self.database_name][self.collection_name].insert_many(
+            documents_to_upsert
+        )
+
+        logger.info("Upsert document successfully")
+
+        return list(chunks.keys())
```

Comment:
The docs say this returns document ids but this is returning the keys from chunks. Is that expected?